addons/website_profile/controllers/main.py

```diff
@@ -37,13 +37,17 @@ def _check_avatar_access(self, user_id, **post):
 
     def _check_user_profile_access(self, user_id):
         user_sudo = request.env['res.users'].sudo().browse(user_id)
+        result = {'denial': False, 'not_found': False, 'user': user_sudo}
         # User can access - no matter what - his own profile
         if user_sudo.id == request.env.user.id:
-            return user_sudo
-        if user_sudo.karma == 0 or not user_sudo.website_published or \
-            (user_sudo.id != request.session.uid and request.env.user.karma < request.website.karma_profile_min):
-            return False
-        return user_sudo
+            return result
+        if not user_sudo.exists():
+            result['not_found'] = True
+        elif user_sudo.karma == 0 or not user_sudo.website_published:
+            result['denial'] = _('This profile is private!')
+        elif request.env.user.karma < request.website.karma_profile_min:
+            result['denial'] = _('Not enough karma to view this profile.')
```

Comment:
Maybe we shouldn't be specific about "this" profile, as it depends on website settings.
```suggestion
            result['denial'] = _('Not have enough karma to view other users' profile.')
```